doc/conf.py

```diff
@@ -706,18 +706,19 @@ def setup(app):
         " non-GUI backend, so cannot show the figure."
     ),
 )
-# Raise warning as error in example to catch warnings when building the documentation
-# Since we are using lock files to build the documentation, we should not have any
-# warnings. Before updating the lock files, we need to fix them.
-for warning_type in (FutureWarning, DeprecationWarning, VisibleDeprecationWarning):
-    warnings.filterwarnings("error", category=warning_type)
-# TODO: remove when pyamg > 5.0.1
-# Avoid a deprecation warning due pkg_resources deprecation in pyamg.
-warnings.filterwarnings(
-    "ignore",
-    message="pkg_resources is deprecated as an API",
-    category=DeprecationWarning,
-)
+if os.environ.get("SKLEARN_DOC_BUILD_WARNINGS_AS_ERRORS", "true").lower() == "true":
```

Comment:
For info, this is breaking the doc build when using python 3.12 (there is a DeprecationWarning: coming from datetime.datetime.utcfromtimestamp)

Reply:
Indeed I think I saw that and left it for later, PR more than welcome :wink:! You would need to add them to the already accepted warnings see  `warnings.filterwarnings("ignore",`

IIRC the deprecation warning is due to python-datetime, it has been fixed in `main` but no release yet, and it seems like it is a bit stuck at the moment